src/TestContext.elm

```diff
@@ -425,6 +468,133 @@ clickButton buttonText testContext =
         testContext
 
 
+{-| Simulates replacing the text in an input field labeled with the given label.
+
+NOTE: Currently, this function requires that you also provide the field id
+(which must match both the `id` attribute of the target `input` element,
+and the `for` attribute of the `label` element).
+After [eeue56/elm-html-test#52](https://github.com/eeue56/elm-html-test/issues/52) is resolved,
+a future release of this package will remove the `fieldId` parameter.
+
+NOTE: TODO: In the future, this will be generalized to work with
+labeled textareas as well as labeled input fields.
+(related to [eeue56/elm-html-test#49](https://github.com/eeue56/elm-html-test/issues/49>))
+
+NOTE: In the future, this will be generalized to work with
+aria accessiblity attributes in addition to working with standard HTML label elements.
+
+If you need to target a `<textarea>` that does not have a label,
+see [`fillInTextarea`](#fillInTextArea).
+
+If you need more control over the finding the target element or creating the simulated event,
+see [`simulate`](#simulate).
+
+-}
+fillIn : String -> String -> String -> TestContext msg model effect -> TestContext msg model effect
+fillIn fieldId label newContent testContext =
+    simulateLabeledInputHelper ("fillIn " ++ toString label)
+        fieldId
+        label
+        [-- TODO: should ensure that known special input types are not set, like `type="checkbox"`, etc?
+        ]
+        (Test.Html.Event.input newContent)
+        testContext
+
+
+{-| Simulates replacing the text in a `<textarea>`.
+
+This function expects that there is only one `<textarea>` in the view.
+If your view has more than one `<textarea>`,
+see [`fillIn`](#fillIn) if your `<textarea>` has an associated `<label>` (preferred),
```

Comment:
this ends up as "If …, then … if …" and gets a little confusing. Maybe:

> If your view has more than one `<textarea>`, assign them `<label>`s (you should do this anyway for accessibility) and use `fillIn`.

Reply:
Fixed!   (Also, I realized that fillIn didn't actually yet work with textareas, so I fixed that too)